backend/fms_core/resources/sample.py

```diff
@@ -135,106 +139,118 @@ def import_obj(self, obj, data, dry_run):
         father = None
 
         if data["Mother ID"]:
-            mother, _ = Individual.objects.get_or_create(
-                name=get_normalized_str(data, "Mother ID"),
-                sex=Individual.SEX_FEMALE,
-                taxon=taxon,  # Mother has same taxon as offspring
-                **({"pedigree": pedigree} if pedigree else {}),  # Mother has same pedigree as offspring
-                **({"cohort": cohort} if cohort else {}),  # Mother has same cohort as offspring TODO: Confirm
-            )
+            try:
+                mother, _ = Individual.objects.get_or_create(
+                    name=get_normalized_str(data, "Mother ID"),
+                    sex=Individual.SEX_FEMALE,
+                    taxon=taxon,  # Mother has same taxon as offspring
+                    **({"pedigree": pedigree} if pedigree else {}),  # Mother has same pedigree as offspring
+                    **({"cohort": cohort} if cohort else {}),  # Mother has same cohort as offspring TODO: Confirm
+                )
+            except ValidationError as e:
+                errors["mother"] = ValidationError(e.messages.pop(), code="invalid")
 
         if data["Father ID"]:
-            father, _ = Individual.objects.get_or_create(
-                name=get_normalized_str(data, "Father ID"),
-                sex=Individual.SEX_MALE,
-                taxon=taxon,  # Father has same taxon as offspring
-                **({"pedigree": pedigree} if pedigree else {}),  # Father has same pedigree as offspring
-                **({"cohort": cohort} if cohort else {}),  # Father has same cohort as offspring TODO: Confirm
-            )
+            try:
+                father, _ = Individual.objects.get_or_create(
+                    name=get_normalized_str(data, "Father ID"),
+                    sex=Individual.SEX_MALE,
+                    taxon=taxon,  # Father has same taxon as offspring
+                    **({"pedigree": pedigree} if pedigree else {}),  # Father has same pedigree as offspring
+                    **({"cohort": cohort} if cohort else {}),  # Father has same cohort as offspring TODO: Confirm
+                )
+            except ValidationError as e:
+                errors["father"] = ValidationError(e.messages.pop(), code="invalid")
 
         # TODO: This should throw a nicer warning if the individual already exists
         # TODO: Warn if the individual exists but pedigree/cohort is different
-        individual, individual_created = Individual.objects.get_or_create(
-            name=get_normalized_str(data, "Individual ID"),
-            sex=get_normalized_str(data, "Sex", default=Individual.SEX_UNKNOWN),
-            taxon=taxon,
-            **({"pedigree": pedigree} if pedigree else {}),
-            **({"cohort": cohort} if cohort else {}),
-            **({"mother": mother} if mother else {}),
-            **({"father": father} if father else {}),
-        )
-        obj.individual = individual
+        try:
```

Comment:
I don't really like the logic here. try get_or_create, if exception then individual_created false..  later on if individual_created is false then.. I think if you want to check whether the individual already exists you should split up here the get_or_create into a get individual, if IndividualDoesNotExist then add warning message, try create individual if exception then error individual.. Seems more straightforward to me